src/vcpkg/vcpkgpaths.cpp

```diff
@@ -368,6 +320,22 @@ namespace vcpkg
             Optional<LockFile> m_installed_lock;
         };
 
+        static Path compute_registries_cache_root(const Filesystem& fs, const VcpkgCmdArguments& args)
+        {
+            Path ret;
+            if (args.registries_cache_dir)
+            {
+                LockGuardPtr<Metrics>(g_metrics)->track_property("X_VCPKG_REGISTRIES_CACHE", "defined");
+                ret = *args.registries_cache_dir;
+            }
+            else
+            {
+                ret = get_platform_cache_home().value_or_exit(VCPKG_LINE_INFO) / "vcpkg" / "registries";
+            }
+
+            return fs.almost_canonical(ret, VCPKG_LINE_INFO);
```

Comment:
We lost all the helpful error messages for `X_VCPKG_REGISTRIES_CACHE` :'(

Reply:
We lost specific messages but I think most all of those already get emitted from the shared bits we're reusing now.

Reply:
I think it would be useful to retain the check that X_VCPKG_REGISTRIES_CACHE must be an absolute path.

Reply:
Why though? This just fixes it with almost_canonical.

Reply:
(Also note now that there's a command line parameter for it so relative paths are more likely and reasonable)

To land this for this tool update I put these checks back but I think we should undo them after discussion once you're back.